File: Aula09/ex01.py
class Triangulo:
  def __init__(self, b, h):
    self.set_base(b)
    self.set_altura(h)
  def get_base(self):
    return self.__b
  def get_altura(self):
    return self.__h
  def set_base(self, b):
    if b >= 0: self.__b = b
    else: raise ValueError
  def set_altura(self, h):
    if h >= 0: self.__h = h
    else: raise ValueError
  def calc_area(self):
    return self.__b * self.__h / 2
  def __str__(self):
    return f"Base = {self.__b}  Altura = {self.__h}  Área = {self.calc_area()}"

x = Triangulo(5, 5)
print(x)

y = Triangulo(10, 20)
print(y)

File: Aula09/Ex01.java
public class Ex01 {
  public static void main(String[] args) {
    Triangulo x = new Triangulo(10, 20);
    //x.b = -10;
    //x.h = -20;
    x.setBase(10);
    x.setAltura(20);
    // Console.WriteLine(x.b);
    System.out.println(x.getBase());
    // Console.WriteLine(x.h);
    System.out.println(x.getAltura());
    System.out.println(x.toString());
    
    Triangulo y = new Triangulo(30, 40);
    System.out.println(y.getBase());
    System.out.println(y.getAltura());
    System.out.println(y);
  }
}

class Triangulo {
  private double b, h;
  public Triangulo(double b, double h) {
    //this.b = b;
    //this.h = h;
    setBase(b);
    setAltura(h);
  }
  public double getBase() {
    return b;
  }
  public double getAltura() {
    return h;
  }
  public void setBase(double b) {
    if (b >= 0) this.b = b;
    else throw new IllegalArgumentException();
  }
  public void setAltura(double h) {
    if (h >= 0) this.h = h;
    else throw new IllegalArgumentException();
  }
  public double calcArea() {
    return b * h / 2;
  }
  public String toString() {
    return String.format("Base = %f  Altura = %f  Área = %f", b, h, calcArea());
  }  
}
